torch/csrc/jit/passes/quantization.cpp

```diff
@@ -818,13 +871,142 @@ void InsertObserversHelper::insertObservers(
+
+  // if the current graph is the entry point graph(the forward graph
+  // of the top level module), we can insert observers in the graph
+  if (!is_entry_point) {
+    for (auto* v : graph->inputs()) {
+      graph_inputs_outputs.insert(v);
+      graph_input_observers.push_back(getObserverFor(v));
+    }
+
+    for (auto* v : graph->outputs()) {
+      graph_inputs_outputs.insert(v);
+      graph_output_observers.push_back(getObserverFor(v));
+    }
+  }
+
+  // We'll return this information to the call site of the graph
+  // since observers should be inserted at call site
+  auto graph_observers = std::make_tuple(graph_input_observers, graph_output_observers);
+
+  // This means the graph is been processed before, we just
+  // need to attach observer modules and construct the information
+  // needed by call site here
+  if (graph_observer_map_.count(graph.get())) {
+    // instance clone of observer module and setAttr
+    for (const auto& observer_attrs : graph_observer_map_.at(graph.get())) {
+      const auto& name = std::get<0>(observer_attrs);
+      const auto& observer = std::get<1>(observer_attrs);
+      module._ivalue()->setAttr(name, observer.clone_instance()._ivalue());
+    }
+    for (auto& invoked_method : getInvokedMethods(module, method_name)) {
+      auto& invoked_module = std::get<0>(invoked_method);
+      const auto& invoked_method_name = std::get<1>(invoked_method);
+      insertObservers(invoked_module, invoked_method_name);
+    }
+    return graph_observers;
+  }
+  GRAPH_DUMP("inserting observer for:", graph);
 
-  // Actually add observer nodes.
-  for (Value* v : values_to_observe) {
-    insertObserverFor(v, module, getObserverModuleFor(v, qconfig));
+  std::stack<Block*> blocks_to_visit;
+  blocks_to_visit.push(graph->block());
+  auto* self = graph->inputs()[0];
+  // We first construct a map from value to the module, then
+  // insert observers for them later, this is to avoid interference
+  // of the inserted observers with the analysis to decide where
+  // to insert observers, also we only insert observers for
+  // "intermediate values" that is not the input/output of the
+  // graph
+  std::unordered_map<Value*, script::Module> values_to_observe;
+  while (!blocks_to_visit.empty()) {
+    Block* b = blocks_to_visit.top();
+    blocks_to_visit.pop();
+    for (Node* n : b->nodes()) {
+      if (observer_nodes_.count(n)) {
+        continue;
+      }
+      if (n->kind() == prim::CallMethod) {
+        auto m = getInvokedModule(module, n, self);
+        auto input_output_observers = insertObservers(m, n->s(attr::name));
+        auto input_observers = std::get<0>(input_output_observers);
+        auto output_observers = std::get<1>(input_output_observers);
+        for (auto i = 0; i < n->inputs().size(); ++i) {
+          if (input_observers[i] && !graph_inputs_outputs.count(n->inputs()[i]) && !values_to_observe.count(n->inputs()[i])) {
```

Comment:
Clang-format :-p

Reply:
will setup clang-format in my editor, seems didn't succeed last time..